plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/expression/SupportsPushdown.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.clickhouse.expression;
+
+import com.clickhouse.data.ClickHouseColumn;
+import com.clickhouse.data.ClickHouseDataType;
+import io.trino.plugin.base.expression.ConnectorExpressionRule.RewriteContext;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcTypeHandle;
+import io.trino.spi.TrinoException;
+import io.trino.spi.expression.Variable;
+
+import java.util.function.BiPredicate;
+
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+
+public class SupportsPushdown
+        implements BiPredicate<Variable, RewriteContext<?>>
+{
+    @Override
+    public boolean test(Variable variable, RewriteContext<?> context)
```

Comment:
What if we could have it as a static method - in `ClickhouseClient` or some util class and use Lambda to pass them to `matching` and use it other places ?